internal/controller/components/modelregistry/modelregistry_controller.go

```diff
@@ -76,14 +75,10 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		// namespaces that may not be known when the controller is started, hence
 		// it should be watched dynamically
 		WatchesGVK(gvk.ServiceMeshMember, reconciler.Dynamic()).
```

Comment:
We probably need to check for this only if service mesh is enabled, previously,ServiceMesh was a pre-condition, now it is optional so this woudl probably fail if the CRD is not present.